yaptide/batch/result_sender.py

```diff
@@ -31,42 +31,63 @@ def send_simulation_results(output_Path: Path, simulation_id: int, update_key: s
+    dict_to_send = {"sim_id": simulation_id, "job_state": simulation_state, "update_key": update_key}
+    jobs_url = f"{backend_url}/jobs"
+    context = ssl.SSLContext()
+
+    req = request.Request(jobs_url,
+                          json.dumps(dict_to_send).encode(), {'Content-Type': 'application/json'},
+                          method='POST')
+    try:
+        with request.urlopen(req, context=context) as res:  # skipcq: BAN-B310
+            if res.getcode() != 202:
+                logging.warning("Sending update to %s failed", jobs_url)
+    except Exception as e:  # skipcq: PYL-W0703
+        logging.error("Sending update to %s failed: %s", jobs_url, str(e))
 
 
 if __name__ == "__main__":
     signal.signal(signal.SIGUSR1, signal.SIG_IGN)
 
-    logging.basicConfig(
-        level=logging.INFO,
-        format="%(asctime)s %(levelname)s %(message)s",
-        handlers=[
-            logging.StreamHandler()
-        ]
-    )
+    logging.basicConfig(level=logging.INFO,
+                        format="%(asctime)s %(levelname)s %(message)s",
+                        handlers=[logging.StreamHandler()])
     parser = argparse.ArgumentParser()
-    parser.add_argument("--output_dir", type=str)
-    parser.add_argument("--sim_id", type=int)
-    parser.add_argument("--update_key", type=str)
-    parser.add_argument("--backend_url", type=str)
+    parser.add_argument("--sim_id", type=int, required=True)
+    parser.add_argument("--update_key", type=str, required=True)
+    parser.add_argument("--backend_url", type=str, required=True)
+    parser.add_argument("--output_dir", type=str, required=False)
+    parser.add_argument("--simulation_state", type=str, required=False)
     args = parser.parse_args()
-    logging.info("output_dir %s", args.output_dir)
+
     logging.info("sim_id %s", args.sim_id)
     logging.info("update_key %s", args.update_key)
     logging.info("backend_url %s", args.backend_url)
 
-    send_simulation_results(
-        output_Path=Path(args.output_dir),
-        simulation_id=args.sim_id,
-        update_key=args.update_key,
-        backend_url=args.backend_url
-    )
+    if args.output_dir:
+        logging.info("output_dir %s", args.output_dir)
+        send_simulation_results(output_Path=Path(args.output_dir),
+                                simulation_id=args.sim_id,
+                                update_key=args.update_key,
+                                backend_url=args.backend_url)
+    elif args.simulation_state:
+        logging.info("No output_dir provided, sending state %s", args.simulation_state)
+        send_simulation_state_update(simulation_id=args.sim_id,
+                                     update_key=args.update_key,
+                                     backend_url=args.backend_url,
+                                     simulation_state=args.simulation_state)
+    else:
+        logging.error("output_dir or simulation_state needs to be provided")
```

Comment:
Right now this script handles both sending simulation results and updating simulation state. From its name it's not obvious what is its goal.

I suggest to rename it and add some docstring which explains where to use which option.
Its not clear from the code that output_dir is needed when script works like results sender and simulation_state is needed when it works like state updater. 

Think of a better naming or separate the functionality into two scripts. The purpose of the code/script should be straightforward and follow the naming